airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -228,6 +260,27 @@ def supports_incremental(self) -> bool:
         """
         return len(self._wrapped_cursor_field()) > 0
 
+    @property
+    def supports_checkpointing(self) -> bool:
+        """
+        :return: True if this stream allows the checkpointing of sync progress and can resume from it on subsequent attempts.
+        This differs from supports_incremental because certain kinds of streams like those supporting resumable full refresh
+        can checkpoint progress in between attempts for improved fault tolerance. However, they will start from the beginning
+        on the next sync job.
+        """
+        if hasattr(type(self), "parent"):
+            # We temporarily gate substream to not support RFR because puts a pretty high burden on connector developers
```

Comment:
I see 3 paths forward:
1. Accept substreams are not resumable
2. Implement support for RFR and substreams in concurrent
3. Implement a something similar to SubstreamPartitionRouter for non-low code connectors

I believe (2) and (3) would be a similar level of effort so I have a preference for (2).

Reply:
Yep I am in agreement as well. This will be part of the concurrent RFR prototype, which conveniently the Stripe endpoint we want to implement is a substream

https://github.com/airbytehq/airbyte-internal-issues/issues/7390